docker/pypi/wmagent/init.sh

```diff
@@ -425,6 +427,39 @@ agent_resource_control() {
     echo "-------------------------------------------------------"
 }
 
+agent_resource_opp() {
+    local stepMsg="Performing $FUNCNAME"
+    echo "-------------------------------------------------------"
+    echo "Start: $stepMsg"
+    if _init_valid $wmaInitResourceOpp && \
+       _init_valid $wmaInitSqlDB
+    then
+        echo "$FUNCNAME: Agent Opportunistic Resource control has been populated already."
+    else
+        echo "$FUNCNAME: triggered."
+        local errVal=0
+        ### Populating opportunistic resource-control
+        echo "$FUNCNAME: Populating opportunistic resource-control"
+        for res in ${!RESOURCE_*}
```

Comment:
Can you please add a comment above this line saying what is expected to be in RESOURCE_* env variables?

In addition, I am confused with the exclamation mark in this line. What is actually happening here? It iterates over all RESOURCE_* env vars that are not empty?

Reply:
Yes, and "!" is needed to actually access the information stored in `RESOURCE_*` variables as `res`